File: src/com/tbilisi/bus/BusListAdapter.java
package com.tbilisi.bus;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;
import android.widget.TextView;

import java.util.ArrayList;

public class BusListAdapter extends BaseAdapter {
    public ArrayList<BusInfo> busList;
    private Context context;

    public BusListAdapter(Context context, ArrayList<BusInfo> busList) {
        this.context = context;
        this.busList = busList;
    }

    public void update(ArrayList<BusInfo> newList) {
        busList = newList;
        notifyDataSetChanged();
    }

    @Override
    public int getCount() {
        return busList.size();
    }

    @Override
    public Object getItem(int i) {
        return busList.get(i);
    }

    @Override
    public long getItemId(int i) {
        return (long) i;
    }

    @Override
    public View getView(int i, View convertView, ViewGroup viewGroup) {
        View view = convertView;
        BusInfo item = busList.get(i);
        LayoutInflater inflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);

        if(view == null) {
            view = inflater.inflate(R.layout.bus_info, viewGroup, false);
        }

        TextView busNumber = (TextView) view.findViewById(R.id.busNumber);
        TextView busDestination = (TextView) view.findViewById(R.id.busDestination);
        TextView busArrival = (TextView) view.findViewById(R.id.busArrival);

        busNumber.setTypeface(A.typeface);
        busDestination.setTypeface(A.typeface);
        busArrival.setTypeface(A.typeface);

        busNumber.setText("№" + String.valueOf(item.number));
        busDestination.setText(item.destination);
        busArrival.setText(String.valueOf(item.arrival) + "წთ");

        return view;
    }

    @Override
    public boolean isEmpty() {
        return busList.isEmpty();
    }
}

File: src/com/tbilisi/bus/util/BusListAdapter.kt
package com.tbilisi.bus.util

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import android.widget.TextView

import com.tbilisi.bus.R
import com.tbilisi.bus.data.BusInfo

import java.util.ArrayList

public class BusListAdapter(private val context: Context, public var busList: ArrayList<BusInfo>?) : BaseAdapter() {

    public fun update(newList: ArrayList<BusInfo>?) {
        busList = newList!!
        notifyDataSetChanged()
    }

    override fun getCount(): Int {
        return busList!!.size()
    }

    override fun getItem(i: Int): Any {
        return busList!!.get(i)
    }

    override fun getItemId(i: Int): Long {
        return i.toLong()
    }

    override fun getView(i: Int, convertView: View?, viewGroup: ViewGroup): View {
        var view: View? = convertView
        val item = busList!!.get(i)
        val inflater = context.getSystemService(Context.LAYOUT_INFLATER_SERVICE) as LayoutInflater

        if (view == null) {
            view = inflater.inflate(R.layout.bus_info, viewGroup, false)
        }

        val busNumber = view!!.findViewById(R.id.busNumber) as TextView
        val busDestination = view.findViewById(R.id.busDestination) as TextView
        val busArrival = view.findViewById(R.id.busArrival) as TextView

        busNumber.text = item.number.toString()
        busDestination.text = item.destination
        busArrival.text = item.arrival.toString() + "წთ"

        return view
    }

    override fun isEmpty(): Boolean {
        return busList!!.isEmpty()
    }
}
